src/SFA.DAS.AANHub.Application/LeavingReasons/Queries/GetLeavingReasons/GetLeavingReasonsQueryResult.cs

```diff
@@ -0,0 +1,65 @@
+﻿using SFA.DAS.AANHub.Domain.Entities;
+
+namespace SFA.DAS.AANHub.Application.LeavingReasons.Queries.GetLeavingReasons;
+
+public class GetLeavingReasonsQueryResult
+{
+    public IEnumerable<LeavingReasonModel> LeavingReasons { get; set; } = null!;
+
+    public static implicit operator GetLeavingReasonsQueryResult(List<LeavingReason> leavingReasons) => new()
+    {
+        LeavingReasons = leavingReasons.Select(r => new LeavingReasonModel(r.Id, r.Category, r.Description, r.Ordering))
+    };
+
+    public LeavingReasonsProcessed ProcessedResult
+    {
+        get
+        {
+            var leavingCategories = new List<LeavingCategory>();
+            var categories = LeavingReasons.Select(r => r.Category).Distinct();
+            foreach (var category in categories.OrderBy(c => c))
+            {
+                var leavingReasons = new List<LeavingReasonProcessed>();
+                foreach (var r in LeavingReasons.Where(l => l.Category == category))
+                {
+                    leavingReasons.Add(new LeavingReasonProcessed
+                    {
+                        Id = r.Id,
+                        Description = r.Description,
+                        Ordering = r.Ordering
+                    });
+                }
+
+                leavingCategories.Add(new LeavingCategory
+                {
+                    Category = category,
+                    LeavingReasons = leavingReasons
+                });
+
+            }
```

Comment:
This logic should be in the handler

Reply:
The logic can be simplified, using LINQ
`LeavingReasons.GroupBy(l => l.Category).Select(g => new LeavingCategory { Category = g.Key, LeavingReasons = g.ToList() }).ToList();`